pkg/client/simple/vfsclientset/clientset.go

```diff
@@ -84,6 +93,73 @@ func (c *VFSClientset) GetFederation(name string) (*kops.Federation, error) {
 	return c.federations().Get(name, metav1.GetOptions{})
 }
 
+func (c *VFSClientset) SecretStore(cluster *kops.Cluster) (fi.SecretStore, error) {
+	configBase, err := registry.ConfigBase(cluster)
+	if err != nil {
+		return nil, err
+	}
+	basedir := configBase.Join("secrets")
+	return secrets.NewVFSSecretStore(basedir), nil
+}
+
+func (c *VFSClientset) KeyStore(cluster *kops.Cluster) (fi.CAStore, error) {
+	configBase, err := registry.ConfigBase(cluster)
+	if err != nil {
+		return nil, err
+	}
+	basedir := configBase.Join("pki")
+	return fi.NewVFSCAStore(basedir), nil
+}
+
+func DeleteAllClusterState(basePath vfs.Path) error {
+	paths, err := basePath.ReadTree()
+	if err != nil {
+		return fmt.Errorf("error listing files in state store: %v", err)
+	}
+
+	for _, path := range paths {
+		relativePath, err := vfs.RelativePath(basePath, path)
+		if err != nil {
+			return err
+		}
+		if relativePath == "config" || relativePath == "cluster.spec" {
+			continue
+		}
+		if strings.HasPrefix(relativePath, "addons/") {
+			continue
+		}
+		if strings.HasPrefix(relativePath, "pki/") {
+			continue
+		}
+		if strings.HasPrefix(relativePath, "secrets/") {
+			continue
+		}
+		if strings.HasPrefix(relativePath, "instancegroup/") {
+			continue
+		}
+
+		return fmt.Errorf("refusing to delete: unknown file found: %s", path)
```

Comment:
`	return fmt.Errorf("refusing to delete: unknown file found: %s", path)`

how about

`	return fmt.Errorf("unknown file found %q, not deleting: %q", path, basePath)`

Reply:
This is the existing code, the point is that we do a `kops delete cluster` and we refuse to delete _anything_, because we can't prove it safe.  That's what the first message is trying to imply (and I think the proposed alternative does not capture)

Reply:
the current message makes sense 👍 

Disregard my comment